part1_pseudocode.txt

```diff
@@ -0,0 +1,22 @@
+
+Write argparse code to input filename/location as a variable.
+
+Create two empty lists - 
+	-One will contain the alignment positions.
+	-The other will contain UMI/Randomers.
+
+Open file:
+
+	Iterate through the file line by line AKA read by read
+	
+	For each read:
+		-Parse the cigar string and adjust the alignment position if soft clipping is present.
+
+		-Once alignment is adjusted, check first list to see if position has been seen before.
+			if not, add it to the list and return true, else return false
+
+		-check to see if UMI/Randomer has been seen before.
+			if not, add it to the list and return true, else return false
+		
+		-if both alignment and UMI functions return true, output read to new file, else go to next read.
```

Comment:
I think I know what you meant, but the way you wrote out your logic it would have some errors:
- Position not seen before: Added to the list, return true
- UMI seen before: Don't add to list, return false
-> Go to next read.

However you're storing a position from a read you're not keeping. Just the way you wrote it, but I'd just make sure that you're running a check for both to return True, then adding them to the lists.

Also consider what exactly you'd write out as a function, such as the cigar string parsing, etc.

Otherwise looks good!